packages/api-server/src/methods/modules/gw.ts

```diff
@@ -448,3 +485,27 @@ function formatHexNumber(
 
   return num.toLowerCase();
 }
+
+function serializeExecuteRawL2TxParameters(
+  serializeRawL2Tx: HexString,
+  blockNumber: HexNumber | null | undefined
+): HexString {
+  const toSerializeObj = {
+    serializeRawL2Tx,
+    blockNumber: blockNumber || "0x",
+  };
+  return JSON.stringify(toSerializeObj);
+}
+
+function getExecuteRawL2TxCacheKey(
+  serializeRawL2Tx: string,
+  tipBlockHash: HexString,
+  memPoolStateRoot: HexString
+) {
+  const hash = "0x" + keccakFromString(serializeRawL2Tx).toString("hex");
```

Comment:
Why using keccak here ?

Reply:
the cache key will include a fingerprint of parameters, serializeRawL2Tx might be very long, so use a hash function here

Reply:
Is blake2b hash better ? It's tx's `gw_tx_hash` .

Reply:
> Is blake2b hash better ? It's tx's `gw_tx_hash` .

I don't have preferred. but since gw_tx_hash is meaningful, so should be better choice

Reply:
actually the parameter is not serializeRawL2Tx but with extra blockNumber params, but I have update to blake2b to make it more compatible with gw toolchain anyway.